torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -194,133 +218,134 @@ def _clear_cache(self) -> None:
+                        device=self._default_device,
                     )
-
-                    self._all_states.append(
-                        _recursive_copy_to_device(replica_state, non_blocking=True, device=torch.device("cpu"))
+                    self._all_state_dicts.append(
+                        _recursive_copy_to_device(local_state_dict, non_blocking=True, device=torch.device("cpu"))
                     )
             else:
-                # Acknowledge broadcasts, and send this rank's shard when needed
-                # Default to CPU space to gain some memory headroom
                 if rank == self.rank:
-                    # Send the state to the reference replica
+                    # Send the optimizer state to the target rank
                     _ = _broadcast_object(
                         self.local_state_dict(),
                         src_rank=self.global_rank,
                         group=self.group,
-                        dist_device=self._device,
+                        device=self._default_device,
                     )
-
                 elif rank != to:
-                    # Discard this tensor/rank, broadcast was being use for compatibility reasons
+                    # Discard the received object; `broadcast()` is used for
+                    # compatibility reasons
                     _ = _broadcast_object(
                         empty_messenger,
                         src_rank=global_rank,
                         group=self.group,
-                        dist_device=self._device,
+                        device=self._default_device,
                     )
 
     def partition_parameters(self) -> List[List[Dict]]:
         r"""
         Partitions parameters across distributed data parallel ranks.
 
         Returns:
-            a list of ``param_groups`` (which is a list of dict) where each
-            element of the list contains the param_groups for a rank. Element 0
-            corresponds to rank 0, etc. We need all the ranks for the broadcast
-            inside ``step()``.
-
-        NOTE: `test_sharding()` and `test_add_param_group()` rely on this
-        function using the sorted-greedy algorithm for partitioning. If the
-        algorithm is changed, please re-examine those two tests in
-        `test_zero_redundancy_optimizer.py` accordingly.
+            A :class:`list` of ``param_groups`` (which is a :class:`list` of
+            :class:`dict`) where each element of the list contains the
+            ``param_groups`` for a rank. Element 0 corresponds to rank 0, etc.
+            Each rank stores the ``param_groups`` for all of the ranks for the
+            collective communication in :meth:`step`.
+
+        NOTE: ``test_sharding()`` and ``test_add_param_group()`` in
+        ``test_zero_redundancy_optimizer.py`` rely on this function using the
+        sorted-greedy algorithm for partitioning. If the algorithm is changed,
+        please re-examine those two tests.
```

Comment:
We shouldn't have notes about test methods in the docs of public methods. Also, shouldn't this method be private? Do we actually expect users to call this method? If not, it should be private.